### PR TITLE
feat(color): add prevfg,prevbg as color specifiers based on the previous foreground/background colors respectively

### LINKED ISSUE #528: Powerline prompt
## Feature Request

I would like to have a prompt with PowerLine characters to make it beautiful

#### Is your feature request related to a problem? Please describe.

I tried with this config

```toml
add_newline = false

[character]
symbol = ""

[git_status]
show_sync_count = true
prefix = " "
suffix = " "
style = "bg:red fg:white"
```

But it is not what i expected

image

#### Describe the solution you'd like

Maybe i will have to wait this issues to be solved

https://github.com/starship/starship/pull/378

https://github.com/starship/starship/issues/346

https://github.com/starship/starship/issues/243

#### Describe alternatives you've considered

Nerd Fonts shows a prompt with PowerLine on [the home page](https://www.nerdfonts.com/)
[](https://github.com/ryanoasis/nerd-fonts/blob/gh-pages/assets/img/nerd-fonts-powerline-extra-terminal.png)

[BobTheFish](https://github.com/oh-my-fish/theme-bobthefish) is a theme that use PowerLine
[](https://github.com/oh-my-fish/theme-bobthefish)

### PR DESCRIPTION
**Pull request by vladimir-lu**
#### Description

This PR adds the `prevfg` and `prevbg` specifiers that are allowed to appear in `fg:` or `bg:` style elements. The value will be based on the previous segment's `AnsiString`. This allows powerline-like separators to have the correct colors when so configured.

#### Motivation and Context

I am migrating from [powerline](https://github.com/powerline/powerline) and one of the features of the prompt strings there is pretty filled status segments. This PR is based off a suggestion in #528 to allow the previous foreground/background color to be available in style specifiers. This PR does not enable all the functionality of Powerline (e.g. directory separators, pipestatus colors would need to be enhanced) but is crucial to allowing the specification of segment styles depending on the previous segment. It is also a much smaller change than other approaches would suggest e.g. #1235

Closes #528

#### Screenshots (if appropriate):

image

#### How Has This Been Tested?

- [x] Add unit tests

- [x] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by vladimir-lu** [comment]
https://github.com/vladimir-lu/starship-theme-powerline is an example config that can be used for manual testing

File: src/module.rs [hunk 1]
```diff
@@ -209,7 +209,7 @@ where
             }
             _ => {
                 used += segment.width_graphemes();
-                current.push(segment.ansi_string());
+                current.push(segment.ansi_string(current.last().map(|s| s.style_ref())));
```
Comment (davidkna) [inline src/module.rs:212]: I think it would be simpler to always store the last `Style` in a variable.

**Review by davidkna — CHANGES_REQUESTED** [review]
Sorry for not reviewing this earlier, but this PR required more attention compared to other PRs.
I found a few issues, but the changes work surprisingly well with fairly few code changes, thank you for contributing this!

**Comment by CEbbinghaus** [comment]
This looks awesome. Glad to see that we are finally getting this functionality. Comments seem reasonable so is there anything preventing this from progressing further?

**Comment by MartenBE** [comment]
Is there any update on this? This would be very handy to change from powerline to starship

**Pull request closed without merging**